lib/shopify-cli/core/monorail.rb

```diff
@@ -46,7 +46,7 @@ def now_in_milliseconds
 
         # we only want to send Monorail events in production or when explicitly developing
         def enabled?
-          Context.new.system? || ENV["MONORAIL_REAL_EVENTS"] == "1"
+          (Context.new.system? || ENV["MONORAIL_REAL_EVENTS"] == "1") && !Context.new.ci?
```

Comment:
If you switch the order (as suggested below) in `prompt_for_consent`, then this change in `enabled?` isn't needed.